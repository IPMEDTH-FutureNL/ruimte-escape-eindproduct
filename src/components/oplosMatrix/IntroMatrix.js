import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";

import "../../css/start.css"
import Astronaut from "../../img/gameplayImages/characters/character1_standard_mouthclosed.svg"

import TextCloud from '../../img/gameplayImages/snippets/textcloud.png'

const IntroMatrix = () => {
    const [text, setText] = useState("We zijn opgestegen " + localStorage.getItem("playerOne") +  " en " + localStorage.getItem("playerTwo") + "! Wat goed dat ik jullie hier tegen kom!");
    const [count, setCount] = useState(0);
    const navigate = useNavigate();

    const startPuzzel = () => {
        navigate('/puzzelTwee');
    }


    const story = (newText) => {
        setCount(count + newText);

        switch(count){
            case 0:
                setText("Het lukt mij niet om de deur te openen. Hiervoor moet ik een code invullen.")
                break;
            case 1:
                setText("Kunnen jullie mij hierbij helpen?");
                break;
            case 2:
                let matrixSection = document.getElementById('container__gameplay-matrix');
                let circleMatrix = document.getElementById("circle-matrix");

                matrixSection.style.display = "none";
                circleMatrix.style.display = "block";
                break;
        }
    }
    return(
        <section className="container__intro-oplosMatrix">
            <div className="circle-matrix" id="circle-matrix" onClick={startPuzzel}></div>

            <section className="container__gameplay-matrix" id="container__gameplay-matrix">
                    <div className="container__astronaut">
                        <div className="astronaut">
                            <img src={Astronaut} alt="astronaut" className="astronaut__image" draggable="false"></img>
                        </div>
                        <div className="textcloud">
                            <img src={TextCloud} alt="tekst balloon" className="textcloud__image" draggable="false"></img>
                            <div className="textcloud__field">
                                <p className="textcloud__text">{text}</p>
                                <button className="continue" onClick={() => story(1)}>Verder</button>
                            </div>
                        </div>
                    </div>
            </section>
        </section>
    )
}

export default IntroMatrix;