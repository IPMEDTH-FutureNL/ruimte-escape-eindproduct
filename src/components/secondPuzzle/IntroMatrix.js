import React from 'react'
import { useNavigate } from "react-router-dom";

const IntroMatrix = () => {

    const navigate = useNavigate();

    const startPuzzel = () => {
        navigate("/puzzelTwee");
    }

    return(
        <section className="container__intro-oplosMatrix">
            <div className="circle-matrix" onClick={startPuzzel}></div>
        </section>
    )
}

export default IntroMatrix;