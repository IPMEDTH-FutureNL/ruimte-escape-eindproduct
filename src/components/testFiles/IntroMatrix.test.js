import renderer from 'react-test-renderer';
import IntroMatrix from "../oplosMatrix/IntroMatrix";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if Overgang Puzzel component is rendered', () => {
    const component = renderer.create(
        <Router>
            <IntroMatrix></IntroMatrix>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})