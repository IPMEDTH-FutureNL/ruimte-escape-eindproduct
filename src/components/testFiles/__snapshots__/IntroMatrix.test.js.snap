// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Overgang Puzzel component is rendered 1`] = `
Array [
  <section
    className="container__intro-oplosMatrix"
  >
    <div
      className="circle-matrix"
      id="circle-matrix"
      onClick={[Function]}
    />
    <section
      className="container__gameplay-matrix"
      id="container__gameplay-matrix"
    >
      <div
        className="container__astronaut"
      >
        <div
          className="astronaut"
        >
          <img
            alt="astronaut"
            className="astronaut__image"
            src="character1_standard_mouthclosed.svg"
          />
        </div>
        <div
          className="textcloud"
        >
          <img
            alt="tekst balloon"
            className="textcloud__image"
            src="textcloud.png"
          />
          <div
            className="textcloud__field"
          >
            <p
              className="textcloud__text"
            >
              Ahh null en null wat goed dat ik jullie hier tegen kom!
            </p>
            <svg
              className="continue"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 256 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </div>
        </div>
      </div>
    </section>
  </section>,
  ",",
]
`;
